lib/taiko.js

```diff
@@ -373,6 +373,38 @@ const _go = async (delta, options = {timeout: 30000}) =>{
     await waitForNavigation(options.timeout, promises).catch(handleTimeout(options.timeout));
 };
 
+/**
+ * Returns window's location attributes as requested.
+ * @param {String} locationAttr - Provide the location attribute to fetch specific value.
+ * @value {"protocol"} http, https
+ * @value {"host"} abcd.com
+ * @value {"pathname"} /mypage
+ * @value {"search"} ?q=test
+ * @value {"toString()"} http://abcd.com/mypage?q=test
+ *
+ * @example
+ * browserURL('protocol');
+ * returns http or https
+ *
+ * @returns {Promise<String>}
+ */
+module.exports.browserURL = async (locationAttr = 'toString()') => {
+    validate();
+    if (['protocol', 'host', 'pathname', 'search', 'toString()'].includes(locationAttr)){
```

Comment:
I recommend simplifying this and only returning the string.
Once the string is returned the user can use the standard library.
https://developer.mozilla.org/en-US/docs/Web/API/URL to their hearts content.

Reply:
recommend: Renaming the function to currentURL makes more meaning when we have multiple tabs open and also aligns with current existing API

Reply:
@zabil agree, but I was just thinking 'toString()' returns the complete URL anyway. Now there are lot of cases where we need to assert on protocol or pathname etc and I thought it will give a faster and easier way for QAs to write tests, saving the time to write codes mentioned in https://developer.mozilla.org/en-US/docs/Web/API/URL.

Thoughts?

Reply:
From the QA perspective it makes sense. 

But Taiko tries hard to be the API as flat as possible so that it's discoverable and usable from the REPL something a user can see immediately. 

The other thing is compatibility with something like chai https://www.chaijs.com/plugins/chai-url/ which works on strings to provide the assertion.

```
expect(currentURL()).to.contain.path('/foo');
```
Becomes readable out of the box.

Hope this makes sense.

Reply:
Ah okay, got it

Reply:
Made the changes @zabil and @saikrishna321

Reply:
@zabil aren't we tightly coupling with chai for assertions now. What as use if I use expect.js for my assertions. In this case, i can't do  `expect(currentURL()).to.contain.path('/foo');`

Should we just return the Object with matches https://developer.mozilla.org/en-US/docs/Web/API/URL#Properties and let the user decides.